ocaml/xapi/bios_strings.ml

```diff
@@ -32,26 +78,49 @@ let get_bios_string name =
     else str
   with _ -> ""
 
+let get_baseboard_strings () =
+  let keys = ["baseboard-manufacturer"; "baseboard-product-name";
+              "baseboard-version"; "baseboard-serial-number"] in
+
+  let output = get_output_of_type "2" in
+  let records = match Angstrom.parse_string P.record_list output with
+  | Ok records -> records
+  | Error msg -> []
+  in
+  let values = match List.length records = 0 with
```

Comment:
Should this be the other way around because currently we try to run List.hd if the list is empty which will fail
We could
```
match records with
| [record] -> record.values
| record::_ -> warning more records (if this is unexpected); record.values
| [] -> warning no records (if this is unexpected); []
```